java-squid/src/main/java/org/sonar/java/se/SymbolicValueRelation.java

```diff
@@ -34,73 +38,154 @@ protected SymbolicValueRelation(SymbolicValue v1, SymbolicValue v2) {
     this.v2 = v2;
   }
 
-  public Boolean impliedBy(List<SymbolicValueRelation> knownRelations) {
+  @Override
+  public int compareTo(SymbolicValueRelation o) {
+    final int[] key1 = key();
+    final int[] key2 = o.key();
+    int answer = Integer.compare(key1[0], key2[0]);
+    if (answer == 0) {
+      answer = Integer.compare(key1[1], key2[1]);
+      if (answer == 0) {
+        answer = o.getClass().getSimpleName().compareTo(o.getClass().getSimpleName());
```

Comment:
so at the end it's only a question of name of class if both symbolic values are the equivalent?! I'm not sure that it's a good approach.